src/common/summarize-profile.js

```diff
@@ -110,17 +97,43 @@ export function summarizeProfile(profile: Profile) {
  */
 function functionNameCategorizer() {
   const cache = new Map();
+
+  // Compile set of categories into a single regular expression that produces
+  // the pattern (whether it's a prefix, stem, substring, or exact match).
+
+  const patternToCategory = new Map();
+
+  const matchPatterns = {
+    prefix: [],
+    stem: [],
+    substring: [],
+    exact: []
+  };
+
+  for (const [matchType, pattern, category] of categories) {
+    patternToCategory.set(pattern, category);
+    let pat = pattern.replace(/\W/g, '\\$&');
+    if (matchType == 'stem')
+      pat += "\\b(?=\\()?";
```

Comment:
I feel like `stem` exists to match the full function name, so maybe it's just enough to add `\\b` at the end of the subpattern in the regexp "constructor" below instead of adding a new pattern. What do you think ?

Also I think an optional lookahead assertion is just a noop anyway ? IIANW it's not even captured, right ?

Lastly I didn't really understand why you've put this here instead of in the regexp "constructor" below ?